qualtran/bloqs/basic_gates/hadamard.py

```diff
@@ -102,8 +123,76 @@ def _hadamard() -> Hadamard:
     return hadamard
 
 
-_HADAMARD_DOC = BloqDocSpec(
-    bloq_cls=Hadamard,
-    import_line='from qualtran.bloqs.basic_gates import Hadamard',
-    examples=[_hadamard],
-)
+_HADAMARD_DOC = BloqDocSpec(bloq_cls=Hadamard, examples=[_hadamard], call_graph_example=None)
+
+
+@frozen
+class CHadamard(Bloq):
+    r"""The controlled Hadamard gate
+
+    Registers:
+        ctrl: The control qubit.
+        q: The target qubit.
+    """
+
+    @cached_property
+    def signature(self) -> 'Signature':
+        return Signature.build(ctrl=1, q=1)
+
+    def decompose_bloq(self) -> 'CompositeBloq':
+        raise DecomposeTypeError(f"{self} is atomic")
+
+    def adjoint(self) -> 'Bloq':
+        return self
+
+    def my_tensors(
+        self, incoming: Dict[str, 'ConnectionT'], outgoing: Dict[str, 'ConnectionT']
+    ) -> List['qtn.Tensor']:
+        import quimb.tensor as qtn
+
+        unitary = np.eye(4, dtype=np.complex128).reshape((2, 2, 2, 2))
+        # Use these inds orderings to set the block where ctrl=1 to the desired gate.
+        inds = [
+            (outgoing['ctrl'], 0),
+            (outgoing['q'], 0),
+            (incoming['ctrl'], 0),
+            (incoming['q'], 0),
+        ]
+        unitary[1, :, 1, :] = _HADAMARD
+
+        return [qtn.Tensor(data=unitary, inds=inds, tags=[str(self)])]
+
+    def as_cirq_op(
+        self, qubit_manager: 'cirq.QubitManager', ctrl: 'CirqQuregT', q: 'CirqQuregT'
+    ) -> Tuple[Union['cirq.Operation', None], Dict[str, 'CirqQuregT']]:
+        import cirq
+
+        (ctrl,) = ctrl
+        (q,) = q
+        return cirq.H.on(q).controlled_by(ctrl), {'ctrl': np.array([ctrl]), 'q': np.array([q])}
+
+    def _t_complexity_(self) -> 'TComplexity':
+        # This is based on the decomposition provided by `cirq.decompose_multi_controlled_rotation`
+        # which uses three cirq.MatrixGate's to do a controlled version of any single-qubit gate.
+        # The first MatrixGate happens to be a clifford, Hadamard operation in this case.
+        # The other two are considered 'rotations'.
```

Comment:
Please link https://github.com/quantumlib/Qualtran/issues/237 in the comments so we can update the T-complexity at a later point.